src/main/scala/sangria/marshalling/circe.scala

```diff
@@ -56,16 +56,14 @@ object circe {
 
     def isDefined(node: Json) = !node.isNull
     def getScalarValue(node: Json) =
-      if (node.isBoolean)
-        node.asBoolean.get
-      else if (node.isNumber) {
-        val num = node.asNumber.get
-        
-        (num.toBigInt orElse num.toBigDecimal).get
-      } else if (node.isString)
-        node.asString.get
-      else
-        throw new IllegalStateException(s"$node is not a scalar value")
+      node.fold(
+        jsonNull = None,
+        jsonBoolean = Some(_),
+        jsonNumber = num => num.toBigInt orElse num.toBigDecimal,
+        jsonString = Some(_),
+        jsonArray = _ => None,
+        jsonObject = _ => None
+      ).getOrElse(throw new IllegalStateException(s"$node is not a scalar value"))
```

Comment:
I like this change! 

My concern is that it has unnecessary boxing in `Option`, though current version has it as well. `getScalarValue` would be This method is a very scalar value in the input, I think it would be better to avoid boxing in this case. 

What do you think about something like this?

```scala
def getScalarValue(node: Json) = {
  def invalidScalar = throw new IllegalStateException(s"$node is not a scalar value")
  
  node.fold(
    jsonNull = invalidScalar,
    jsonBoolean = identity,
    jsonNumber = num ⇒ num.toBigInt orElse num.toBigDecimal getOrElse invalidScalar,
    jsonString = identity,
    jsonArray = _ ⇒ invalidScalar,
    jsonObject = _ ⇒ invalidScalar
  )
}
```